pandas/core/arrays/sparse/array.py

```diff
@@ -1305,24 +1307,16 @@ def astype(self, dtype: AstypeArg | None = None, copy: bool = True):
         future_dtype = pandas_dtype(dtype)
         if not isinstance(future_dtype, SparseDtype):
             # GH#34457
-            if isinstance(future_dtype, np.dtype):
-                values = np.array(self)
-                return astype_nansafe(values, dtype=future_dtype)
-            else:
-                # pylint: disable-next=used-before-assignment
-                dtype = cast(ExtensionDtype, dtype)
-                cls = dtype.construct_array_type()
-                return cls._from_sequence(self, dtype=dtype, copy=copy)
+            values = np.asarray(self)
```

Comment:
Is this robust if the original type was `pd.SparseDtype(pd.Int64Dtype)` with an `NA` and `astype(pd.Int32Dtype)` was called?

Reply:
Oh looks like there an existing bug here too (missing import)

```
In [1]: arr = pd.array([1, pd.NA], dtype=pd.SparseDtype(pd.Int64Dtype()))

In [2]: arr
Out[2]:
[1, <NA>]
Fill: <NA>
IntIndex
Indices: array([0], dtype=int32)

In [3]: arr.dtype
Out[3]: Sparse[Int64, <NA>]

In [4]: arr.astype(pd.Int16Dtype())
NameError: name 'ExtensionDtype' is not defined
```

Reply:
Could you raise a `NotImplementedError` for the EADtypes here for now then?

Reply:
id rather do that as separately to Do It Right along with a couple other bugs mentioned in #49631